modules/kvformat/filterx-func-parse-kv.c

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright (c) 2024 shifter
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx-func-parse-kv.h"
+#include "filterx/object-string.h"
+#include "filterx/object-primitive.h"
+#include "filterx/expr-literal.h"
+#include "filterx/filterx-eval.h"
+#include "filterx/filterx-globals.h"
+#include "filterx/object-json.h"
+#include "filterx/object-message-value.h"
+#include "filterx/object-null.h"
+
+#include "kv-parser.h"
+#include "parser/parser-expr.h"
+#include "scratch-buffers.h"
+
+#define PARSE_KV_OPTS_COUNT 4
+
+enum parse_kv_opts
+{
+  PARSE_KV_OPTS_PREFIX = 0,
+  PARSE_KV_OPTS_VALUE_SEP = 1,
+  PARSE_KV_OPTS_PAIR_SEP = 2,
+  PARSE_KV_OPTS_STRAY_WORDS_VAL_NAME = 3,
+};
+
+static const gchar *parse_kv_opts_names[PARSE_KV_OPTS_COUNT] =
+{
+  "prefix",
```

Comment:
I don't think we need a prefix option here. we return as a JSON anyway, so the results will be encapsulated nicely and filterx can be used to move it around.

with only name-value pairs prefix was used for namespacing, but that's not required anymore.